### DISCUSSION TITLE
OSDquery: filter by benchmark series

### DISCUSSION THREAD

**Discussion by latenooker** [ncss-tech/soilDB#234] [Ideas]
Is it currently possible to limit an OSD query to benchmark series within soilDB as allowed by the [NRCS OSD Query Facility](https://soilseries.sc.egov.usda.gov/osdquery.aspx)? I didn't see a way to do that within the SoilWeb OSD search interface.

**Comment by brownag** [comment]
Thanks for using the "Discussions" feature!

Currently the `OSDquery()` function and the SoilWeb interface do not have an argument to filter benchmark soils, but this would be a very useful feature. @dylanbeaudette can correct me but it should be straightforward to add on the SoilWeb API and then update the interfaces to allow toggling that filter.

In the meantime here is a workaround that downloads a snapshot of the Series Classification database from https://github.com/ncss-tech/SoilWeb-data/ and uses the `benchmarksoilflag` column to get a list of benchmark soils. This list can then be used in a second step to filter out your OSDquery results as needed.

``` r
library(aqp)
#> This is aqp 1.40
library(soilDB)
# library(R.utils)

# download series classification database and unzip
x <- download.file("https://github.com/ncss-tech/SoilWeb-data/raw/main/files/SC-database.csv.gz",
                   destfile = 'series-classification-database.csv.gz', mode = "wb")
R.utils::gunzip('series-classification-database.csv.gz', overwrite = TRUE)
sc <- read.csv('series-classification-database.csv')

# subset to just benchmark soils
sc.benchmark <- subset(sc, benchmarksoilflag)

# find OSDs of Haploxeralfs in MLRAs 17 and 18
mlra <- c("17","18")
osd_haploxeralfs <- do.call('rbind', lapply(mlra, function(x) OSDquery(mlra = x, taxonomic_class = "Haploxeralfs")))

# get just benchmark
osd_benchmark_haploxeralfs <- subset(osd_haploxeralfs, osd_haploxeralfs$series %in% sc.benchmark$soilseriesname)

# inspect result
o <- fetchOSD(osd_benchmark_haploxeralfs$series)
plotSPC(o)
```

![](https://i.imgur.com/Qp5eM32.png)

**Comment by dylanbeaudette** [comment]
EDIT: haha, @brownag beat me to it.

It isn't currently possible, but something that I would like to add in the near future. Thanks for the idea. Until this is implemented, you can augment your search with the Soil Classification database snapshot. This contains a column called `benchmarksoilflag` that can be used to refine the results from `OSDquery()`.

```r
tf <- tempfile()
download.file('https://github.com/ncss-tech/SoilWeb-data/raw/main/files/SC-database.csv.gz', tf)
x <- read.csv(tf)

# see "benchmarksoilflag" column
x[x$soilseriesname == 'CORNING', 1:4]
```

**Comment by latenooker** [comment]
Thanks for the solution, @brownag and @dylanbeaudette! I appreciate the quick replies
